pkg/master/master.go

```diff
@@ -261,9 +265,10 @@ func New(c *Config) *Master {
 		v1beta3:               c.EnableV1Beta3,
 		nodeIPCache:           NewIPCache(c.Cloud, util.RealClock{}, 30*time.Second),
 
-		masterCount:     c.MasterCount,
-		readOnlyServer:  net.JoinHostPort(c.PublicAddress, strconv.Itoa(int(c.ReadOnlyPort))),
-		readWriteServer: net.JoinHostPort(c.PublicAddress, strconv.Itoa(int(c.ReadWritePort))),
+		masterCount:           c.MasterCount,
+		readOnlyServerIpPort:  net.JoinHostPort(c.PublicAddress, strconv.Itoa(int(c.ReadOnlyPort))),
+		readWriteServerIpPort: net.JoinHostPort(c.PublicAddress, strconv.Itoa(int(c.ReadWritePort))),
```

Comment:
Using the same IP for 2 services is going to confuse someone, probably DNS, maybe more.  You need two IPs for now.